### ISSUE TITLE
Inconsistent return values in `auth` functions

### ISSUE THREAD

**Issue by der-gabe** [SUSE/aimaas#115]
The function `authenticated_user` returns a `User`, while `authorized_user` returns a `Callable`, which in turn returns a `User`.

Such inconsistencies make both contributions and reviews more difficult and should be cleaned up.

**Comment by esskayesss** [comment]
i agree with this issue. i have also encountered other instances where the way things have been named can end up being very confusing.
